pkg/server/manifest/utils.go

```diff
@@ -0,0 +1,250 @@
+package manifest
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/tensorleap/leap-cli/pkg/log"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+type GithubTag struct {
+	Name string `json:"name"`
+}
+
+func GetTensorleapRepoFileContent(filePath, ref string) ([]byte, error) {
+
+	owner := "tensorleap"
+	repo := "helm-charts"
+	url := fmt.Sprintf("https://api.github.com/repos/%s/%s/contents/%s?ref=%s", owner, repo, filePath, ref)
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("failed to retrieve file: %s", resp.Status)
+	}
+
+	var fileData struct {
+		Content string `json:"content"`
+	}
+	err = json.NewDecoder(resp.Body).Decode(&fileData)
+	if err != nil {
+		return nil, err
+	}
+
+	decodedContent, err := base64.StdEncoding.DecodeString(fileData.Content)
+	if err != nil {
+		return nil, err
+	}
+
+	return decodedContent, nil
+}
+
+func getK3sImages(k3sVersion string) ([]string, error) {
+	url := fmt.Sprintf("https://github.com/k3s-io/k3s/releases/download/%s/k3s-images.txt", strings.Replace(k3sVersion, "-", "+", 1))
+	images, err := getImagesFromUrl(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed fetching latest k3s images: %v", err)
+	}
+
+	return images, nil
+}
+
+func getImagesFromUrl(url string) ([]string, error) {
+	b, err := getFileByUrl(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get images list: %v", err)
+	}
+	return getImagesFromBytes(b)
+}
+
+func getFileByUrl(url string) ([]byte, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		return nil, fmt.Errorf("failed getting file from (%s): %v", url, resp.StatusCode)
+	}
+	imagesFile, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return imagesFile, nil
+}
+
+func getImagesFromBytes(imagesFile []byte) ([]string, error) {
+
+	imagesFromFile := strings.Split(string(imagesFile), "\n")
+	images := []string{}
+	for _, image := range imagesFromFile {
+		if len(image) > 0 {
+			images = append(images, image)
+		}
+	}
+	return images, nil
+}
+
+var ErrNoTags = fmt.Errorf("no tag found")
+
+func GetLatestTag() (string, error) {
+	repoOwner := "tensorleap"
+	repoName := "helm-charts"
+	page := 1
+	per_page := 100
+
+	for {
+
+		url := fmt.Sprintf("https://api.github.com/repos/%s/%s/tags?page=%v&per_page=%v", repoOwner, repoName, page, per_page)
```

Comment:
Let's consider using the gh CLI package

Reply:
I will use `github.com/cli/go-gh` repo since the `cli` itself is using it and it is hard to use it directly from the code (the function that actually executed the cmd are private - we can work around it I think currently it is better to use graph ql with the helpers of go-gh )